deps/v8/include/v8.h

```diff
@@ -5381,6 +5381,17 @@ class V8_EXPORT Isolate {
   static Isolate* GetCurrent();
 
   /**
+   * Custom callback used by embedders to help V8 determine if it should abort
+   * when it throws and no internal handler can catch the exception.
+   * If FLAG_abort_on_uncaught_exception is true, then V8 will abort if either:
+   * - no custom callback is set.
+   * - the custom callback set returns true.
+   * Otherwise it won't abort.
+   */
+  typedef bool (*abort_on_uncaught_exception_t)(Isolate*);
```

Comment:
I'd make it return an enum so that it can be extended easily in the future.

The naming is kind of incongruous with V8's normal style.  `UncaughtExceptionCallback` is closer to what I'd expect.

For that matter, is it actually necessary to abort inside V8?  You could rewrite the logic in isolate.cc to this:

```c++
if (uncaught_exception_callback_ != nullptr &&
    PredictExceptionCatcher() != CAUGHT_BY_JAVASCRIPT) {
  uncaught_exception_callback_(reinterpret_cast<v8::Isolate*>(this));
}
```

It's then up to the callback to abort or not.  You can optionally pass `message_obj` as the second argument.